dirsrvtests/tests/suites/clu/dsidm_account_test.py

```diff
@@ -120,8 +119,374 @@ def test_dsidm_account_entry_status_with_lock(topology_st, create_test_user):
+        4. Success
+        5. Success
+        6. Success
+        7. Success
+    """
+
+    standalone = topology_st.standalone
+    args = FakeArgs()
+    args.json = False
+    json_list = ['type',
+                 'list',
+                 'items']
+
+    log.info('Empty the log file to prevent false data to check about group')
+    topology_st.logcap.flush()
+
+    log.info('Test dsidm account list without json')
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, check_value=test_user_name)
+
+    log.info('Test dsidm account list with json')
+    args.json = True
+    list(standalone, DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, content_list=json_list, check_value=test_user_name)
+
+    log.info('Delete the account')
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+    test_account.delete()
+
+    log.info('Test empty dsidm account list with json')
+    list(standalone,DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, content_list=json_list, check_value_not=test_user_name)
+
+    log.info('Test empty dsidm account list without json')
+    args.json = False
+    list(standalone,DEFAULT_SUFFIX, topology_st.logcap.log, args)
+    check_value_in_log_and_reset(topology_st, check_value_not=test_user_name)
+
+
+@pytest.mark.xfail(reason='DS6515')
+@pytest.mark.skipif(ds_is_older("1.4.2"), reason="Not implemented")
+def test_dsidm_account_get_by_dn(topology_st, create_test_user):
+    """ Test dsidm account get-by-dn option
+
+    :id: 07945577-2da0-4fd9-9237-43dd2823f7b8
+    :setup: Standalone instance
+    :steps:
+        1. Run dsidm account get-by-dn for an account without json
+        2. Check the output content is correct
+        3. Run dsidm account get-by-dn for an account with json
+        4. Check the output content is correct
+    :expectedresults:
+        1. Success
+        2. Success
+        3. Success
+        4. Success
+    """
+
+    standalone = topology_st.standalone
+    accounts = nsUserAccounts(standalone, DEFAULT_SUFFIX)
+    test_account = accounts.get(test_user_name)
+
+    args = FakeArgs()
+    args.dn = test_account.dn
+    args.json = False
+
+    account_content = ['dn: {}'.format(test_account.dn),
+                       'cn: {}'.format(test_account.rdn),
+                       'displayName: {}'.format(test_user_name),
+                       'gidNumber: 2000',
+                       'homeDirectory: /home/{}'.format(test_user_name),
+                       'objectClass: top',
+                       'objectClass: nsPerson',
+                       'objectClass: nsAccount',
+                       'objectClass: nsOrgPerson',
+                       'objectClass: posixAccount',
+                       'uid: {}'.format(test_user_name),
+                       'uidNumber: 1000']
```

Comment:
Because of this, I think, we need to rewrite `create_test_user` so it creates exactly this specific user. Explicit better than implicit. And `test_user = users.create_test_user()` is too implicit for the new tests.

Reply:
Fixed